### PR TITLE
src: add eslint configuration and npm script

### PR DESCRIPTION
**Pull request by lance**
The linting configuration is based on [`eslint-recommended`](https://eslint.org/docs/user-guide/configuring#using-eslint-recommended) with some customization for arrow functions, double quoted strings and a few other things.

Running `npx standardx --fix` automatically fixed > 2000 issues. The remaining 200+ issues need
to be fixed by hand. Additionally, all strings are double quotes which is not typically standard and I wonder about fixing that too.

This is a potentially large change since it touches most files, marking as WIP.

Fixes: https://github.com/cloudevents/sdk-javascript/issues/54

Signed-off-by: Lance Ball <lball@redhat.com>

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  "env": {
+    "es6": true,
+    "node": true,
+    "mocha": true
+  },
+  "rules": {
+    "space-before-function-paren": ["error", "never"],
+    "standard/no-callback-literal": "off",
+    "arrow-spacing": "error",
+    "arrow-parens": ["error", "as-needed"],
```
Comment (lance) [inline .eslintrc:12]: This [rule](https://eslint.org/docs/rules/arrow-parens#as-needed) should probably be changed as it conflicts with the [codacy linting](https://app.codacy.com/manual/cloudevents/sdk-javascript/pullRequest?prid=5358785) rules.

Comment (fabiojose) [reply .eslintrc:12]: Could you change it to follow codacy rules?

**Comment by lance** [comment]
@fabiojose I am ready to remove the WIP label from this, however it seems that something within `standardx` linting tool is incompatible with Node.js versions 6 and 7. As I mentioned in https://github.com/cloudevents/sdk-javascript/pull/49#issuecomment-614840928 these versions are currently EOL. If #49 is merged, I can rebase and that should solve it.

**Comment by fabiojose** [comment]
@lance, please fix that conflicts.

**Comment by lance** [comment]
@fabiojose conflicts are resolved. This is ready for review.

**Comment by lance** [comment]
@fabiojose ping - any chance this can land soon? There are lots of changes in this PR, and the longer it sits the more difficult it makes landing new PRs.

**Pull request merged**